dogfood/main.tf

```
@@ -34,6 +45,7 @@ resource "coder_agent" "dev" {
     code-server --auth none --port 13337 &
     sudo service docker start
     coder dotfiles -y 2>&1 | tee  ~/.personalize.log
+    "if [ -n ${var.dotfiles_uri} ]; then coder dotfiles var.dotfiles_uri -y 2>&1 | tee  ~/.personalize.log; fi"
```

Okay found [this](https://stackoverflow.com/questions/73775302/terraform-place-variable-inside-eof-tag) and these are called [Heradoc strings](https://developer.hashicorp.com/terraform/language/expressions/strings#heredoc-strings)

Now I'm second guessing it. Is this what it should be?

```suggestion
    if [ -n "${var.dotfiles_uri}" ]; then coder dotfiles "${var.dotfiles_uri}" -y 2>&1 | tee  ~/.personalize.log; fi
```

Not sure if the quotes will stay so could be ""${var.dotfilesa-uri}""

---

@jsjoeio I think a great way to deal with variables is to assign them to a new variable in the scripts, like this:

```sh
DOTFILES_URI=${var.dotfiles_uri}
if [ -n "$DOTFILES_URI" ]; then
   # ...
fi
```

This way escaping the value becomes a non-issue because the assignment supports basically any input.